avax-synapse-stablepool/contracts/Strategy.sol

```diff
@@ -0,0 +1,479 @@
+        return vault.strategies(address(this)).totalDebt;
+    }
+
+    function lpTokenToWant(uint256 tokens) public view returns (uint256) {
+        if (tokens == 0) {
+            return 0;
+        }
+
+        //we want to choose lower value of virtual price and amount we really get out
+        //this means we will always underestimate current assets. 
+        uint256 virtualOut = virtualPriceToWant().mul(tokens).div(1e18);
+
+        uint256 realOut = swapPool.calculateRemoveLiquidityOneToken(tokens, curveId);
+
+        return Math.min(virtualOut, realOut);
+    }
+
+    function virtualPriceToWant() public view returns (uint256) {
+        uint256 virtualPrice = swapPool.getVirtualPrice();
+
+        if (want_decimals < 18) {
+            return virtualPrice.div(10 ** (uint256(uint8(18) - want_decimals)));
+        } else {
+            return virtualPrice;
+        }
+    }
+    
+    function lpTokensInYVault() public view returns (uint256) {
+        uint256 balance = yvToken.balanceOf(address(this));
+
+        if (yvToken.totalSupply() == 0) {
+            //needed because of revert on priceperfullshare if 0
+            return 0;
+        }
+
+        uint256 pricePerShare = yvToken.pricePerShare();
+        //curve tokens are 1e18 decimals
+        return balance.mul(pricePerShare).div(1e18);
+    }
+
+    function nativeToWant(uint256 _amtInWei) public view override virtual returns (uint256) {}
+
+    function estimatedTotalAssets() public view override returns (uint256) {
+        uint256 totalLpTokens = lpTokensInYVault().add(emissionToken.balanceOf(address(this)));
+        return IERC20(targetToken).balanceOf(address(this)).add(lpTokenToWant(totalLpTokens));
+    }
+
+
+    // our main trigger is regarding our DCA 
+    function harvestTrigger(uint256 callCostinEth)
+        public
+        view
+        override
+        returns (bool)
+    {
+        StrategyParams memory params = vault.strategies(address(this));
+
+        // harvest no matter what once we reach our maxDelay
+        if (block.timestamp.sub(params.lastReport) > maxReportDelay) {
+            return true;
+        }
+
+        // trigger if we want to manually harvest
+        if (forceHarvestTriggerOnce) {
+            return true;
+        }
+
+        // trigger if we have enough credit
+        if (vault.creditAvailable() >= minHarvestCredit) {
+            return true;
+        }
+
+        // otherwise, we don't harvest
+        return false;
+    }
+
+    function prepareReturn(uint256 _debtOutstanding)
+        internal
+        override
+        returns (
+            uint256 _profit,
+            uint256 _loss,
+            uint256 _debtPayment
+        )
+    {
+        _debtPayment = _debtOutstanding;
+
+        uint256 debt = vault.strategies(address(this)).totalDebt;
+        uint256 currentValue = estimatedTotalAssets();
+        uint256 wantBalance = IERC20(targetToken).balanceOf(address(this));
```

Comment:
We can't use target token for this, `want` should be used and should never be able to be changed